### ISSUE TITLE
Make bounds check in hyperrectangle constructor optional

### ISSUE THREAD

**Issue by mforets** [JuliaReach/LazySets.jl#2015]
I think it should be "undefined" (can crash, use at your own risk, etc.). Compare to `IntervalArithmetic.interval` which checks that the interval is well formed, and `IntervalArithmetic.Interval` which doesn't peform the check.

```julia
julia> IntervalArithmetic.interval(2, 1)
ERROR: ArgumentError: `[2, 1]` is not a valid interval. Need `a ≤ b` to construct `interval(a, b)`.
Stacktrace:
 [1] interval(::Int64, ::Int64) at /home/mforets/.julia/packages/IntervalArithmetic/lyDPa/src/intervals/intervals.jl:111
 [2] top-level scope at REPL[5]:1

julia> IntervalArithmetic.Interval(2, 1)
[2, 1]

julia> diam(ans)
-1.0
```
This option would be more for debugging purposes, or speedup the creation of a very large number of rectangles (which you know are well defined).

----

Ref: [this comment](https://github.com/sisl/NeuralVerification.jl/pull/88#issuecomment-591409318).

**Comment by tomerarnon** [comment]
Just to note, assuming the underlying functions will maintain their current behavior, this would be the result:

```julia
julia> h = Hyperrectangle([0,0], [1,1])
Hyperrectangle{Int64,Array{Int64,1},Array{Int64,1}}([0, 0], [1, 1])

julia> h.radius .= -1;

julia> radius(h)
1.0

julia> radius_hyperrectangle(h)
2-element Array{Int64,1}:
 -1
 -1

julia> low(h), high(h)
([1, 1], [-1, -1])
```

I.e. the radius is a norm, and therefore positive, but `radius_hyperrectangle` is negative and therefore `low` and `high` are inverted.

**Comment by mforets** [comment]
true, thanks for checking @tomerarnon. in the PR (https://github.com/JuliaReach/LazySets.jl/pull/2018) i wrote "undefined" as i dont know what other strange result may arise :) for instance, i dont think that the support function makes sense

**Comment by tomerarnon** [comment]
Good point. I'm not sure if the following makes any practical sense. For the record:

Regular 0-center 1-radius HR:
```julia
# d => σ(d,h), ρ(d,h)
 [-1, -1] => (σ = [-1, -1], ρ = 2)
  [0, -1] => (σ = [1, -1], ρ = 1) 
  [1, -1] => (σ = [1, -1], ρ = 2) 
  [-1, 0] => (σ = [-1, 1], ρ = 1) 
   [1, 0] => (σ = [1, 1], ρ = 1)  
  [-1, 1] => (σ = [-1, 1], ρ = 2) 
   [0, 1] => (σ = [1, 1], ρ = 1)  
   [1, 1] => (σ = [1, 1], ρ = 2)  
```

negative radius HR:
```julia
 [-1, -1] => (σ = [1, 1], ρ = -2)  
  [0, -1] => (σ = [-1, 1], ρ = -1) 
  [1, -1] => (σ = [-1, 1], ρ = -2) 
  [-1, 0] => (σ = [1, -1], ρ = -1) 
   [1, 0] => (σ = [-1, -1], ρ = -1)
  [-1, 1] => (σ = [1, -1], ρ = -2) 
   [0, 1] => (σ = [-1, -1], ρ = -1)
   [1, 1] => (σ = [-1, -1], ρ = -2)
```

**Comment by mforets** [comment]
yes, the the results do not need to make sense, that's why i discourage to use the flag other than for debugging purposes :)

we merged this option in pr #2018, so its in `master` already and should be available in the next release. it will be a minor version though, since there is some other change which is breaking (https://github.com/JuliaReach/LazySets.jl/wiki/Release-log-tracker).
